drivers/sensor/max30208/max30208.c

```diff
@@ -0,0 +1,346 @@
+		}
+		k_sleep(K_MSEC(MAX30208_POLL_TIME));
+	}
+
+	/* Read one sample */
+	ret = max30208_readout_sample(dev);
+
+	if (ret < 0) {
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+static const struct sensor_driver_api max30208_driver_api = {
+	.sample_fetch = max30208_sample_fetch,
+	.channel_get = max30208_channel_get,
+#ifdef CONFIG_MAX30208_TRIGGER
+	.trigger_set = max30208_trigger_set,
+	.attr_set = max30208_attr_set,
+#endif
+};
+
+static int max30208_init(const struct device *dev)
+{
+	const struct max30208_config *config = dev->config;
+
+	uint8_t part_id;
+	uint8_t sys_ctrl;
+
+	/* Wait for I2C bus */
+	if (!device_is_ready(config->bus.bus)) {
+		LOG_ERR("I2C dev %s not ready", config->bus.bus->name);
+		return -ENODEV;
+	}
+
+	/* Check the part id to make sure this is MAX30208 */
+	int ret = i2c_reg_read_byte_dt(&config->bus, MAX30208_REG_PART_ID, &part_id);
+
+	if (ret < 0) {
+		LOG_ERR("Could not get Part ID");
+		return ret;
+	}
+	if (part_id != MAX30208_PART_ID) {
+		LOG_ERR("Got Part ID 0x%02x, expected 0x%02x", part_id, MAX30208_PART_ID);
+		return -EIO;
+	}
+
+	/* Reset the sensor */
+	ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_SYS_CTRL, MAX30208_RESET_MASK);
+	if (ret < 0) {
+		LOG_ERR("Failed to reset sensor!");
+		return ret;
+	}
+	/* Wait for reset to be cleared */
+	k_sleep(K_MSEC(MAX30208_RESET_TIME));
+	do {
+		ret = i2c_reg_read_byte_dt(&config->bus, MAX30208_REG_SYS_CTRL, &sys_ctrl);
+		if (ret < 0) {
+			LOG_ERR("Couldn't read system control after reset");
+			return ret;
+		}
+	} while (sys_ctrl & MAX30208_RESET_MASK);
+
+	/* Write the FIFO configuration register */
+	ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_FIFO_CFG1, config->fifo.fifo_a_full);
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize FIFO config 1!");
+		return ret;
+	}
+	ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_FIFO_CFG2,
+				    MAX30208_STC_TO_B(config->fifo.config2));
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize FIFO config 2!");
+		return ret;
+	}
+
+	/* Write the GPIO configuration register */
+	ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_GPIO_SETUP,
+				    MAX30208_STC_TO_B(config->gpio_setup));
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize GPIO setup!");
+		return ret;
+	}
+
+#ifdef CONFIG_MAX30208_TRIGGER
+
+	ret = max30208_init_interrupt(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to initialize interrupt!");
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+#ifdef CONFIG_MAX30208_FIFO_ROLLOVER_EN
+#define MAX30208_FIFO_RO .fifo_ro = CONFIG_MAX30208_FIFO_ROLLOVER_EN,
+#else
+#define MAX30208_FIFO_RO .fifo_ro = 0,
+#endif
+
+#ifdef CONFIG_MAX30208_FIFO_A_FULL_TYPE
+#define MAX30208_A_FULL_TYPE .a_full_type = CONFIG_MAX30208_FIFO_A_FULL_TYPE,
+#else
+#define MAX30208_A_FULL_TYPE .a_full_type = 0,
+#endif
+
+#ifdef CONFIG_MAX30208_FIFO_STAT_CLR
+#define MAX30208_FIFO_STAT_CLR .fifo_stat_clr = CONFIG_MAX30208_FIFO_STAT_CLR,
+#else
+#define MAX30208_FIFO_STAT_CLR .fifo_stat_clr = 0,
+#endif
```

Comment:
Kconfig options can be used directly without storing them in config. Why not make these DT properties?, so they can be set per-instance?

Reply:
would you make all of my Kconfig options to DT properties?